java/base-client/src/main/java/no/bankaxept/epayment/client/base/BaseClient.java

```diff
@@ -116,13 +116,25 @@ public Builder withSuppliedToken(Supplier<String> tokenSupplier) {
             return this;
         }
 
-        public Builder withScheduledToken(String username, String password) {
-            this.tokenPublisher = new ScheduledAccessTokenPublisher("/bankaxept-epayment/access-token-api/v1/accesstoken", apimKey, username, password, Clock.systemDefaultZone(), Executors.newScheduledThreadPool(1), httpClient);
+        public Builder withScheduledToken(String id, String secret) {
+            this.tokenPublisher = new ScheduledAccessTokenPublisher.Builder()
+                    .httpClient(httpClient)
+                    .uri("/bankaxept-epayment/access-token-api/v1/accesstoken")
+                    .credentials(id, secret)
+                    .apimKey(apimKey)
+                    .build();
             return this;
         }
 
-        public Builder withScheduledToken(String username, String password, Clock clock) {
-            this.tokenPublisher = new ScheduledAccessTokenPublisher("/bankaxept-epayment/access-token-api/v1/accesstoken", apimKey, username, password, clock, Executors.newScheduledThreadPool(1), httpClient);
+        public Builder withScheduledToken(String id, String secret, Clock clock) {
+            this.tokenPublisher = new ScheduledAccessTokenPublisher.Builder()
+                    .httpClient(httpClient)
+                    .uri("/bankaxept-epayment/access-token-api/v1/accesstoken")
+                    .grantType(GrantType.client_credentials)
+                    .credentials(id, secret)
+                    .apimKey(apimKey)
+                    .clock(clock)
+                    .build();
             return this;
```

Comment:
The one sets grant type, the other not, which is inconsistent. Also, as client_credentials is the only one we support, we don't need to include it in the builder. You could perhaps set it implicitly with the `credentials(String, String)`?

Reply:
I set it implicitly, renamed it to `clientCredentials`, and made `grantType()` private. Does that make sense?

Reply:
Or do we still want to set it manually?